src/udev/udevadm-hwdb.c

```diff
@@ -35,11 +35,69 @@
  * Uses a Patricia/radix trie to index all matches for efficient lookup.
  */
 
-static const char * const conf_file_dirs[] = {
-        UDEV_HWDB_DIR,
-        UDEV_LIBEXEC_DIR "/hwdb.d",
-        NULL
-};
+static ssize_t follow_path (const char *path_variable, char **destination, size_t max) {
```

Comment:
I have the feeling that this can be made simpler by using `strtok_r`.

Note that it modifies the input. But there may be an advantage - in this way all the paths may share a single allocation.